tinker/admin/redirects/__init__.py

```diff
@@ -99,6 +98,67 @@ def new_redirect_submit(self):
                 'message': 'Your redirect is invalid. Make sure the from_path is not "/" and the to_url exists'
             })
 
+    # Saves the edits to an existing redirect
+    @route('/edit-redirect-submit', methods=['post'])
+    def edit_redirect_submit(self):
+        form = self.base.dictionary_encoder.encode(request.form)
+        id = form['edit-id']
+        from_path = form['edit-redirect-from']
+        to_url = form['edit-redirect-to']
+        short_url = form.get('edit-redirect-short-url') == 'true'
+        expiration_date = form.get('edit-expiration-date')
+        username = session["username"]
+        last_edited = datetime.now()
+
+        if expiration_date:
+            if expiration_date == 'None':
+                expiration_date = None
+            elif '-' in expiration_date:
+                expiration_date = datetime.strptime(expiration_date, '%Y-%m-%d')
+            else:
+                expiration_date = datetime.strptime(expiration_date, "%a %b %d %Y")
+        else:
+            expiration_date = None
+
+        if from_path is None or to_url is None:
+            return abort(400)
+
+        if self.base.paths_are_valid(from_path, to_url):
+            if not from_path.startswith("/"):
```

Comment:
should this be in the "paths_are_valid" method?

Reply:
My thought: The "paths_are_valid" method as it is right now is intended to return True or False. This little chunk of code is altering the from_path, so they are doing separate tasks that might be tricky to combine.

Reply:
oh sure. I wasn't looking closely at what this was doing. I think you're right. This is fine here 👍